langchain/src/schema/runnable.ts

```diff
@@ -82,4 +106,401 @@ export abstract class Runnable<
+            ),
+          batchOptions
+        )
+      );
+    } catch (e) {
+      await Promise.all(
+        runManagers.map((runManager) => runManager?.handleChainError(e))
+      );
+      throw e;
+    }
+    await Promise.all(
+      runManagers.map((runManager, i) =>
+        runManager?.handleChainEnd(_coerceToDict(finalOutputs[i], "output"))
+      )
+    );
+    return finalOutputs;
+  }
+
+  async *_streamIterator(
+    input: RunInput,
+    options?: RunnableConfig
+  ): AsyncGenerator<RunOutput> {
+    const callbackManager_ = await CallbackManager.configure(
+      options?.callbacks,
+      undefined,
+      options?.tags,
+      undefined,
+      options?.metadata
+    );
+    const runManager = await callbackManager_?.handleChainStart(
+      this.toJSON(),
+      _coerceToDict(input, "input")
+    );
+    let nextStepInput = input;
+    try {
+      for (const step of [this.first, ...this.middle]) {
+        nextStepInput = await step.invoke(
+          nextStepInput,
+          this._patchConfig(options, runManager?.getChild())
+        );
+      }
+    } catch (e) {
+      await runManager?.handleChainError(e);
+      throw e;
+    }
+    let concatSupported = true;
+    let finalOutput;
+    try {
+      const iterator = await this.last._streamIterator(
+        nextStepInput,
+        this._patchConfig(options, runManager?.getChild())
+      );
+      for await (const chunk of iterator) {
+        yield chunk;
+        if (concatSupported) {
+          if (finalOutput === undefined) {
+            finalOutput = chunk;
+          } else {
+            try {
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              finalOutput = (finalOutput as any).concat(chunk);
+            } catch (e) {
+              finalOutput = undefined;
+              concatSupported = false;
+            }
+          }
+        }
+      }
+    } catch (e) {
+      await runManager?.handleChainError(e);
+      throw e;
+    }
+    await runManager?.handleChainEnd(_coerceToDict(finalOutput, "output"));
+  }
+
+  pipe<NewRunOutput>(
+    coerceable: RunnableLike<RunOutput, NewRunOutput>
+  ): RunnableSequence<RunInput, NewRunOutput> {
+    if (RunnableSequence.isInstance(coerceable)) {
+      return new RunnableSequence({
+        first: this.first,
+        middle: this.middle.concat([
+          this.last,
+          coerceable.first,
+          ...coerceable.middle,
+        ]),
+        last: coerceable.last,
+      });
+    } else {
+      return new RunnableSequence({
+        first: this.first,
+        middle: [...this.middle, this.last],
+        last: _coerceToRunnable(coerceable),
+      });
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  static isInstance(thing: any): thing is RunnableSequence {
+    return Array.isArray(thing.middle) && Runnable.isInstance(thing);
+  }
+
+  static fromRunnables<RunInput, RunOutput>([first, ...runnables]: [
```

Comment:
Rename to `.from()` ? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from